model/task/db.go

```diff
@@ -138,12 +141,35 @@ var (
 	},
 	}
 
-	// This should reflect Task.GetDisplayStatus()
-	addDisplayStatus = bson.M{
+	addUnattainableDepStatus = bson.M{
 		"$addFields": bson.M{
-			DisplayStatusKey: displayStatusExpression,
+			HasUnattainableDepsKey: bson.M{
```

Comment:
One note (but need not block this PR, since I've removed myself): It might be better to use a `$project` for the `$filter` instead of an `$addFields`. If you're never going to use the array of subdocuments again, then there's no need to keep them in the document. Since they can be pretty big, it might be helpful to reduce them to an array of size 0 or 1 immediately instead of keeping them around.

Put another way, your current approach is
1. Add a new field, an array of size either 0 or 1
2. Add a new field based on the size of the new field in #1

Instead, you could
1. Change the current field to an array of size 0 or 1
2. Change the field again to a boolean based on the size of the array

The first approach uses 2 `$addFields`. The latter uses 2 `$project`s.